src/main/java/spoon/support/reflect/declaration/CtRecordImpl.java

```diff
@@ -117,8 +117,8 @@ public <C extends CtType<Object>> C addTypeMemberAt(int position, CtTypeMember m
 			JLSViolation.throwIfSyntaxErrorsAreNotIgnored(this, String.format("%s method is native or abstract, both is not allowed",
 					member.getSimpleName()));
 		}
-		if (member instanceof CtAnonymousExecutable) {
-			JLSViolation.throwIfSyntaxErrorsAreNotIgnored(this, "Anonymous executable is not allowed in a record");
+		if (member instanceof CtAnonymousExecutable && !member.isStatic()) {
+			JLSViolation.throwIfSyntaxErrorsAreNotIgnored(this, "Instance initializer is not allowed in a record");
```

Comment:
An oversight from myself earlier, we should add the JLS paragraph here. Could you add it?